sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/TypedSchemaTransformProvider.java

```diff
@@ -39,7 +42,16 @@
 @Internal
 public abstract class TypedSchemaTransformProvider<ConfigT> implements SchemaTransformProvider {
 
-  protected abstract Class<ConfigT> configurationClass();
+  @SuppressWarnings("unchecked")
+  protected Class<ConfigT> configurationClass() {
+    Optional<ParameterizedType> parameterizedType =
```

Comment:
Since you just put it in an `Optional` and then take it out again, might be simpler to go ahead like

```java
@Nullable ParameterizedType parameterizedType = (ParameterizedType) getClass().getGenericSuperclass();
checkStateNotNull(superClass, "Could not ...");
return (Class<ConfigT>) parameterizedType.getActualTypeArguments[0];
```

FWIW I am not sure if `getActualTypeArguments[0]` could still be a type variable in some cases. You might want to check that it is a usefully defined type in some more ways... but also this is probably just always going to work, because of how these things are authored. Very nice!

Reply:
Thanks, applied the suggestion.